Tests/ConnectionLostRetryPolicyTests.swift

```diff
@@ -0,0 +1,220 @@
+//
+//  ConnectionLostRetryPolicyTests.swift
+//
+//  Copyright (c) 2014-2018 Alamofire Software Foundation (http://alamofire.org/)
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+@testable import Alamofire
+import Foundation
+import XCTest
+
+class ConnectionLostRetryPolicyTestCase: BaseTestCase {
+
+    // MARK: - Helper Types
+
+    private class StubRequest: DataRequest {
+        let urlRequest: URLRequest
+        override var request: URLRequest? { return urlRequest }
+
+        init(_ url: URL, method: HTTPMethod, session: Session) {
+            let request = Session.RequestConvertible(
+                url: url,
+                method: method,
+                parameters: nil,
+                encoding: URLEncoding.default,
+                headers: nil
+            )
+
+            urlRequest = try! request.asURLRequest()
+
+            super.init(
+                convertible: request,
+                underlyingQueue: session.rootQueue,
+                serializationQueue: session.serializationQueue,
+                eventMonitor: session.eventMonitor,
+                interceptor: nil,
+                retryPolicies: [],
+                delegate: session
+            )
+        }
+    }
+
+    // MARK: - Properties
+
+    private let idempotentMethods: Set<HTTPMethod> = [.get, .head, .put, .delete, .options, .trace]
+    private let nonIdempotentMethods: Set<HTTPMethod> = [.post, .patch, .connect]
+    private var methods: Set<HTTPMethod> { return idempotentMethods.union(nonIdempotentMethods) }
+
+    private let session = Session(startRequestsImmediately: false)
+
+    private let url = URL(string: "https://api.example.com")!
+    private let connectionLostError = NSError(domain: URLError.errorDomain, code: URLError.networkConnectionLost.rawValue, userInfo: nil)
```

Comment:
Probably. If I remember right, they're able to bridge between the two, but I'm not 100% on that. Been a while since I worked on the error logic. I'll look into it and update accordingly 👍